Essentials/src/main/java/com/earth2me/essentials/AsyncTimedTeleport.java

```diff
@@ -113,6 +116,26 @@ public void run() {
                     }
                     try {
                         cancelTimer(false);
+
+                        // the target might change location, so check if they can accept teleports.
+                        if (AsyncTeleport.TeleportType.TPA == teleport.getTpType() && timer_teleportTarget instanceof PlayerTarget) {
+                            final Player targetPlayer = Bukkit.getPlayer(((PlayerTarget) timer_teleportTarget).getUUID());
+                            if (targetPlayer == null) return;
+                            try {
+                                if (!ess.getPermissionsHandler().hasPermission(targetPlayer, "essentials.tpaccept")) {
+                                    teleportOwner.sendMessage(tl("teleportNoAcceptPermission", targetPlayer.getDisplayName()));
+                                    return;
+                                }
+                            } catch (Exception ex) {
```

Comment:
is this happening a lot? don't feel like we should be getting exceptions on permission lookups for an online player